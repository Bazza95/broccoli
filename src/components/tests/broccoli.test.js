import React from 'react';
import Broccoli from '../broccoli';
import Registration from '../registration';
import Footer from '../footer';
import { shallow, mount } from 'enzyme';

// Wraps everything so I can target children and mounting in a simpler fashion
describe("Broccoli", () => {
    let state;
    let screen;
    const broccoli = () => {
        screen = mount(
            <Broccoli {...state}/>
        );
        return screen;
    };

    beforeEach(() => {
        state = {
            open: undefined
        };
        screen = undefined;
    });

    // Renders the main page
    describe("`Broccoli` renders", () => {
        it('renders as expected', () => {
            const wrapper = shallow(
                <Broccoli />
            );
            expect(wrapper).toMatchSnapshot();
        });
    });

    // The footer has 0 props
    describe("rendered `Footer`", () => {
        it("does not receive any props", () => {
            const footer = broccoli().find(Footer);
            expect(Object.keys(footer.props()).length).toBe(0);
        });
    });

    // A footer is always rendered
    it("always renders a `Footer`", () => {
        expect(broccoli().find(Footer).length).toBe(1);
    });

    // When the open prop is false, Registration isn't rendered
    describe("when `open` is false", () => {
        beforeEach(() => {
            state.open = false;
        });

        it("sets the rendered `Registration`'s `open` prop to undefined'", () => {
            expect(broccoli().find(Registration).length).toBe(0);
        });
    });
});
