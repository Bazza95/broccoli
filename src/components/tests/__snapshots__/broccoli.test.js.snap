// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Broccoli \`Broccoli\` renders renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Broccoli) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "button": "Broccoli-button-3",
        "mainDash": "Broccoli-mainDash-2",
        "root": "Broccoli-root-1",
        "subHeading": "Broccoli-subHeading-4",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "button": "Broccoli-button-3",
          "mainDash": "Broccoli-mainDash-2",
          "root": "Broccoli-root-1",
          "subHeading": "Broccoli-subHeading-4",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
